### ISSUE TITLE
Force Push always pushes Rules even when Rules are excluded in Push Publishing filter 

### ISSUE THREAD

**Issue by yolabingo** [dotCMS/core#19648]
**Describe the bug**
Rules are included in Push Publishing bundles when using "Force Push Everything" filter, even when excluded in the Force Push filter:
`publishing-filters/ForcePush.yml   excludeDependencyClasses: [Rule]`

**To Reproduce**
1. Exclude "Rule" as dependencies in Push Publishing filters - see [publishing-filters.zip](https://github.com/dotCMS/core/files/5633740/publishing-filters.zip)
2. Create a Rule on PP sender
3. Edit any contentlet on the sender and Push Publish it with "Force Push Everything" filter
4. The Rule will be included in the bundle with the contentlet and Push Published to the receiver

Similarly, deleting the rule on the sender then Force Pushing any content will delete the Rule on the receiver

**Expected behavior**
`ForcePush.yml:  excludeDependencyClasses: [Rule]` should prevent Rules from being added to Push Publish bundles for contentlets

**Screenshots**
https://drive.google.com/file/d/1lii-geQvhXrjOEAIRIxQkD7iBLB0mZ0A/view?usp=sharing

**Additional context**
Tested on 5.3.8.2

**Comment by yolabingo** [comment]
cloud support https://dotcms.zendesk.com/agent/tickets/102905

**Comment by wezell** [comment]
related #19679

**Comment by nollymar** [comment]
PR Enterprise: https://github.com/dotCMS/enterprise/pull/767

**Comment by nollymar** [comment]
**Important to consider:** If you set `excludeDependencyClasses: [Rule]` and the property `PUSH_PUBLISHING_RULES_OVERWRITE=true` (by default, it is set to true), rules will be wipped out after pushing. For that reason, we suggest to turn off `PUSH_PUBLISHING_RULES_OVERWRITE` first.

**Comment by fabrizzio-dotCMS** [comment]
I turned off the flag `PUSH_PUBLISHING_RULES_OVERWRITE=false` on the receiver so Rules can be updated
Also turned had the ForcePush.yml to include Rules  like this:

```
title: Force Push Everything
default: false
roles: DOTCMS_BACK_END_USER
filters:
   ## Lucene query that exclude content to be added to the bundle
   excludeQuery: ""
   ## list of classes that should never be pushed: ContentType, Template, Containers, Folder, Host, Links, Workflow, Language, Rule, Contentlet, Category, User, OSGI
   excludeClasses: []
   ## check for any dependency that the pushed asset may need (if false will not look for any dependency or relationship)
   dependencies: true
   ## Lucene query that exclude content to be added to the bundle as a dependency
   excludeDependencyQuery: ""
   ##  list of classes that should never be pushed as dependencies: ContentType, Template, Containers, Folder, Host, Links, Workflow, Language, Rule, Contentlet, Category
   excludeDependencyClasses: [Rule]
   ## force push as always worked, if true will push regardless the pushed asset history
   forcePush: true
   ## check for relationships (if false neither the Relationship object nor the related content nor the related content type will be pushed)
   relationships: true
```

It is important to notice that there can be files under two different locations
`WEB-INF/publishing-filters` and 
`assets/server/publishing-filters`
Being the latest the one that takes precedence

o this test was carried out modifying the file under `assets/server/publishing-filters`

**Comment by john-thomas-dotcms** [comment]
So what happens if you do push a rule (_not_ filtering it out), and have `PUSH_PUBLISHING_RULES_OVERWRITE=false`?

Does just the rule you pushed get updated/overwritten?

**Comment by nollymar** [comment]
@john-thomas-dotcms when `PUSH_PUBLISHING_RULES_OVERWRITE=false`, existing rules get updated. Otherwise, they are deleted and created again when `PUSH_PUBLISHING_RULES_OVERWRITE=true`

**Comment by bryanboza** [comment]
Fixed, tested on release-21.02 // Postgres // FF
